### ISSUE TITLE
SC3045: POSIX.1-2024 specifies `cd -P -e`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3306]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC3045
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set -e
cd -P -e /
```

#### Here's what shellcheck currently says:
```
Line 3	SC3045: In POSIX sh, cd -e is undefined.
```
#### Here's what I wanted or expected to see:

Nothing.

POSIX.1-2024 specifies `-e` option.
<https://pubs.opengroup.org/onlinepubs/9799919799/utilities/cd.html>

Note that dash v0.5.12-123-g98cc11a5dbbc does not support it yet.
